tools/sync-module-dependencies/internal/helpers/files.go

```diff
@@ -0,0 +1,116 @@
+package helpers
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+
+	"github.com/grafana/replace-generator/internal/types"
+	"gopkg.in/yaml.v3"
+)
+
+type FileHelper struct {
+	// ScriptDir is the directory where the sync-module-dependencies tools are located.
+	// This is where templates and output files are stored.
+	ScriptDir string
+
+	// ProjectRoot is the root directory of the Alloy project.
+	ProjectRoot string
+
+	// ProjectReplacesPath is the absolute path to dependency-replacements.yaml.
+	ProjectReplacesPath string
+}
+
+var fileHelper *FileHelper
+
+func GetFileHelper() *FileHelper {
+	if fileHelper == nil {
+		fileHelper = newFileHelper()
+	}
+	return fileHelper
+}
+
+func newFileHelper() *FileHelper {
+	scriptDir, err := os.Getwd()
+	if err != nil {
+		log.Fatalf("Failed to resolve working directory: %v", err)
+	}
+
+	scriptDir, err = filepath.Abs(scriptDir)
+	if err != nil {
+		log.Fatalf("Failed to resolve script directory: %v", err)
+	}
+
+	projectRoot, err := filepath.Abs(filepath.Join(scriptDir, "..", ".."))
+	if err != nil {
+		log.Fatalf("Failed to resolve project root: %v", err)
+	}
+
+	projectReplacesPath := filepath.Join(projectRoot, "dependency-replacements.yaml")
+
+	absReplacesPath, err := filepath.Abs(projectReplacesPath)
+	if err != nil {
+		log.Fatalf("Failed to resolve dependency-replacements.yaml: %v", err)
+	}
+
+	return &FileHelper{
+		ScriptDir:           scriptDir,
+		ProjectRoot:         projectRoot,
+		ProjectReplacesPath: absReplacesPath,
+	}
+}
+
+func (d *FileHelper) TemplatePath(fileType types.FileType) string {
+	var templateName string
+	switch fileType {
+	case types.FileTypeMod:
+		templateName = "replaces-mod.tpl"
```

Comment:
Do we have this file type switch here so we can add an OCB YAML file support in the future?